File: Programmers/[43162] 네트워크/KingDonggyu/P_43162.swift
import Foundation

func solution(_ n:Int, _ computers:[[Int]]) -> Int {
    var answer: Int = 0
    var visited: [Bool] = Array(repeating: false, count: n)
    
    func dfs(_ x: Int) {
        visited[x] = true
        for i in 0..<n {
            if !visited[i] && computers[x][i] == 1 {
                dfs(i)
            }
        }
    }
    
    for i in 0..<n {
        if !visited[i] {
            dfs(i)
            answer += 1
        }
    }
    
    return answer
}

File: Programmers/[43162] 네트워크/KingDonggyu/P_43162.py
def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]
    
    def dfs(x):
        visited[x] = True
        for i in range(n):
            if not visited[i] and computers[x][i]:
                dfs(i)
    
    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer += 1
    
    return answer